packages/platforms/browser/lib/routing-provider/react-router-routing-provider.ts

```diff
@@ -0,0 +1,32 @@
+import { type OnSettle } from '../on-settle'
+import { createDefaultRoutingProvider } from './default-routing-provider'
+import { stringToRegexp } from './string-to-regexp'
+
+export interface RouteObject {
+  path?: string
+  children?: RouteObject[]
+}
+
+function flattenRoutes (routes: RouteObject[], _prefix: string = ''): string[] {
+  const prefix = `${!_prefix || _prefix === '/' ? _prefix : `${_prefix}/`}`
+  return [
+    ...routes.map(route => `${prefix}${route.path || ''}`),
+    ...routes.reduce<string[]>(
+      (accum, route) => [...accum, ...(route.children ? flattenRoutes(route.children, `${prefix}${route.path}`) : [])],
+      []
+    )
+  ]
+}
+
+export const createReactRouterRoutingProvider = (onSettle: OnSettle, location: Location) => {
```

Comment:
For react-router `createBrowserRouter`, yes. You need to pass in all your routes when you create the router and there is no mechanism to change it. Vue router has a concept of dynamic routes, which is why in that PR I've opted to accept the router itself, so `router.getRoutes()` can be called to avail ourselves of any newly added (or removed) routes since the router was initially created.